modules/video-recorder-core/src/main/java/com/selenide/videorecorder/VideoRecorderScreenShot.java

```diff
@@ -0,0 +1,169 @@
+ */
+public class VideoRecorderScreenShot extends TimerTask {
+  private static final Logger log = LoggerFactory.getLogger(VideoRecorderScreenShot.class);
+
+  static final String ffmpeg = "%s -loglevel error -f image2pipe -avioflags direct " +
+    "-fpsprobesize 0 -probesize 32 -analyzeduration 0 -c:v mjpeg -i - -y -an -r 24 -c:v vp8 -qmin 0 -qmax 50 " +
+    "-crf 8 -deadline realtime -speed 8 -b:v 1M -threads 1 %s";
+
+  static final String defaultRecordsFolder = "build/records";
+
+  BrowsingContext browsingContext;
+
+  final String ffmpegExecutable = Loader.load(org.bytedeco.ffmpeg.ffmpeg.class);
+
+  private String defaultVideoName;
+
+  private Process recordProcess;
+  private OutputStream outputStream;
+  private Path pathToSaveVideo;
+
+  CaptureScreenshotParameters captureScreenshotParameters;
+
+  public VideoRecorderScreenShot(WebDriver webDriver) {
+    this(webDriver, null, null);
+  }
+
+  public VideoRecorderScreenShot(WebDriver webDriver, String className, String testName) {
+    try {
+      prepareRecordsFolder();
+      createDirectoryToSaveVideoForTest(className, testName);
+      defaultVideoName = new Date().getTime() + ".webm";
+      initVideoRecordingProcess(webDriver);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void createDirectoryToSaveVideoForTest(String className, String testName) throws IOException {
+    pathToSaveVideo = Path.of(defaultRecordsFolder);
+
+    if (!StringUtils.isEmpty(testName)) {
+      pathToSaveVideo = pathToSaveVideo.resolve(className);
+    }
+    if (!StringUtils.isEmpty(className)) {
+      pathToSaveVideo = pathToSaveVideo.resolve(testName);
+    }
+    Files.createDirectories(pathToSaveVideo);
+  }
+
+  private void initVideoRecordingProcess(WebDriver driver) throws IOException, InterruptedException {
+    captureScreenshotParameters = new CaptureScreenshotParameters();
+    captureScreenshotParameters
+      .origin(CaptureScreenshotParameters.Origin.VIEWPORT)
+      .imageFormat("image/png", 0.5);
+    log.debug("Init BiDi");
+    browsingContext = new BrowsingContext(driver, driver.getWindowHandle());
+    recordProcess = Runtime.getRuntime().exec(ffmpeg.formatted(ffmpegExecutable,
+      Path.of(pathToSaveVideo.toString(), defaultVideoName).toFile().getAbsolutePath()));
+    outputStream = recordProcess.getOutputStream();
+    Thread.sleep(1000);
+  }
+
+
+  @Override
+  public void run() {
+    log.debug("Video recording");
+    try {
+      log.debug("Make screenshot");
+      byte[] lastImage = Base64.getDecoder().decode(browsingContext.captureScreenshot(captureScreenshotParameters));
+      byte[] convertedImage = convertPngToJpeg(lastImage);
+      log.debug("Write screenshot to stream");
+      for (int i = 0; i < 24; i++) {
+        outputStream.write(convertedImage);
+      }
+    } catch (Exception e) {
+      log.debug(e.getMessage());
+    }
+  }
+
+  private byte[] convertPngToJpeg(byte[] pngAsByteArray) {
```

Comment:
We don't need it. Ffmpeg can join png too. By converting you just make GC do extra work 😊

Reply:
Yes, you are right. I've deleted this piece of code already.